src/basic/src/SequenceOfVectors.C

```diff
@@ -1536,6 +1538,93 @@ SequenceOfVectors<V,M>::unifiedWriteContents(
   }
 
   if (m_env.inter0Rank() >= 0) {
+    if (fileType == UQ_FILE_EXTENSION_FOR_HDF_FORMAT) {
+#ifdef QUESO_HAS_HDF5
+      unsigned int chainSize = this->subSequenceSize();
+      unsigned int numParams = m_vectorSpace.dimLocal();
+      double * data;
+      data = (double *)malloc(numParams * chainSize * sizeof(double));
+      queso_require_msg(data, "couldn't allocate memory for data");
+
+      for (unsigned int i = 0; i < chainSize; ++i) {
+        V tmpVec(*(m_seq[i]));
+        for (unsigned int j = 0; j < numParams; ++j) {
+          data[numParams*i+j] = tmpVec[j];
+        }
+      }
+
+      int numChains = m_env.inter0Comm().NumProc();
+      int numrecv = numParams * chainSize * numChains;
+      double * recvbuf = (double *)malloc(numrecv * sizeof(double));
+
+      m_env.inter0Comm().template Gather<double>(data,
+                                                 numParams * chainSize,
+                                                 recvbuf,
+                                                 numParams * chainSize,
+                                                 0,
+                                                 "",
+                                                 "");
+
+      if (m_env.inter0Rank() == 0) {
```

Comment:
This looks like the right thing to do for now, but in the long run phdf5 support would be nice.